cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMetadataConfigurationState.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+/**
+ * Represents a desired configuration state needed on nodes (brokers and/or controllers) when transitioning
+ * across states in the FSM
+ */
+public enum KafkaMetadataConfigurationState {
+
+    /**
+     * Full ZooKeeper. Brokers should have ZooKeeper only.
+     */
+    ZK,
+
+    /**
+     * Full ZooKeeper. Controllers should have KRaft, ZooKeeper and migration enabled
+     */
+    PRE_MIGRATION,
+
+    /**
+     * Migration going on. Both brokers and controllers should have KRaft, ZooKeeper and migration enabled.
+     */
+    MIGRATION,
+
+    /**
+     * Finalising migration. Brokers don't have ZooKeeper anymore. Controllers still configured with it.
+     */
+    POST_MIGRATION,
+
+    /**
+     * Full KRaft. Both brokers and controllers have KRaft only.
+     */
+    KRAFT;
+
+    /**
+     * @return if the Kafka metadata are fully stored in ZooKeeper
+     */
+    public boolean isZooKeeper() {
+        return ZK.equals(this);
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is in pre-migration
+     *         Controllers should have KRaft, ZooKeeper and migration enabled
+     */
+    public boolean isPreMigration() {
+        return PRE_MIGRATION.equals(this);
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is in migration
+     *         Both brokers and controllers should have KRaft, ZooKeeper and migration enabled
+     */
+    public boolean isMigration() {
+        return MIGRATION.equals(this);
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is in post-migration
+     *         Brokers don't have ZooKeeper anymore. Controllers still configured with it
+     */
+    public boolean isPostMigration() {
+        return POST_MIGRATION.equals(this);
+    }
+
+    /**
+     * @return if the Kafka metadata are fully stored in KRaft
+     */
+    public boolean isKRaft() {
+        return KRAFT.equals(this);
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is from ZooKeeper-based up to KRaft migration (and dual-write) going on
+     */
+    public boolean isZooKeeperOrMigration() {
+        return this.ordinal() <= MIGRATION.ordinal();
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is from ZooKeeper-based up to KRaft post-migration
+     */
+    public boolean isZooKeeperOrPostMigration() {
+        return this.ordinal() <= POST_MIGRATION.ordinal();
+    }
+
+    /**
+     * @return if the Kafka metadata configuration state is from KRaft pre-migration up to the full KRaft-based
+     */
+    public boolean isPreMigrationOrKRaft() {
```

Comment:
It's have been on my todo list for weeks but with low priority. At this point, it's time to rename all methods :-) 
I can confirm they will be renamed by replacing `Or` with `To` because it covers a range of states.
These methods were added to avoid exposing `if` conditions where states checks were needed but just using a simple method and "understandable" call (but to be really "understandable I have to change the names ;-)).

Reply:
I think the methods are fine and they indeed simplify the code readability. Just need the name to correspond to what the yactually return :-D.